modules/omx/module.c

```diff
@@ -0,0 +1,149 @@
+#include "omx.h"
+
+#include <stdlib.h>
+
+#include <re/re.h>
+#include <rem/rem.h>
+#include <baresip.h>
+
+/////// baresip Compatibility Layer
+
+int omx_vidisp_alloc(struct vidisp_st **vp, const struct vidisp* vd, struct vidisp_prm *prm,
+	const char *dev, vidisp_resize_h *resizeh, void *arg);
+int omx_vidisp_display(struct vidisp_st *st, const char *title,
+		   const struct vidframe *frame);
+int omx_vidisp_update(struct vidisp_st *st, bool fullscreen, int orient, const struct vidrect* window);
+void omv_vidisp_hide(struct vidisp_st *st);
+
+
+
+struct vidisp_st {
+	const struct vidisp *vd;  /* inheritance */	
+	struct vidsz size;             /**< Current size          */
+	struct omx_state* omx;
+};
+
+static struct vidisp* vid; // ???
+
+static struct omx_state omx;
+
+static void destructor(void *arg)
+{
+	struct vidisp_st *st = arg;
+	omx_display_disable(st->omx);
+}
+
+int omx_vidisp_alloc(struct vidisp_st **vp, const struct vidisp* vd, struct vidisp_prm *prm,
+	const char *dev, vidisp_resize_h *resizeh, void *arg)
+{
+	info("vidisp_alloc\n");
+	
+	struct vidisp_st *st;
+
+	/* Not used by OMX */
+	(void) prm;
+	(void) dev;
+	(void) resizeh;
+	(void) arg;
+	
+	st = mem_zalloc(sizeof(*st), destructor);
+	if (!st)
+		return ENOMEM;
+
+	st->vd = vd;
+	*vp = st;
+	
+	st->omx = &omx;
+
+	return 0;
+}
+
+
+
+/////// BARESIP BOILERPLATE
+
+int omx_vidisp_display(struct vidisp_st *st, const char *title,
+		   const struct vidframe *frame)
+{	
+	//info("vc4 display\n");
+	
+	if (frame->fmt != VID_FMT_YUV420P) {
+		return EINVAL;
+	}
+	
+	// Get omx buffer handle
+	
+	// GHOST omx_copy_vidframe_to_buf()
+	
+	// FUNCTION check_resize
+	
+	if (memcmp(&st->size, &frame->size, sizeof(struct vidsz))) {
+		info("new frame size: w=%d h=%d\n", frame->size.w, frame->size.h);
+		info("linesize[0]=%d\tlinesize[1]=%d\tlinesize[2]=%d\n", frame->linesize[0], frame->linesize[1], frame->linesize[2]);
+		omx_display_enable(st->omx, frame->size.w, frame->size.h, frame->linesize[0]);
+		memcpy(&st->size, &frame->size, sizeof(struct vidsz));
+	}
+	
+	// Get Buffer
+	int i;
+	void* buf;
+	uint32_t len;
+	uint32_t offset = 0;
+	omx_display_input_buffer(st->omx, &buf, &len);
+	
+	for (i = 0; i < 3; i++) {
+		size_t plane_bytes = frame->linesize[i] * frame->size.h / (i > 0 ? 2 : 1);
+		void* dest = buf + offset;
+		
+		if (offset + plane_bytes > len) {
+			warn("Too large frame for OMX Buffer size. Expected %d, got %d\n", len, offset + plane_bytes);
+			return -ENOMEM;
+		}
+		
+		memcpy(dest, frame->data[i], plane_bytes);
+		offset += plane_bytes;
+	}
+	omx_display_flush_buffer(st->omx);	
+	return 0;
+}
+
+int omx_vidisp_update(struct vidisp_st *st, bool fullscreen, int orient, const struct vidrect* window)
+{
+	warn("omx_vidisp_update\n");
+	printf("fullscreen=%d	orient=%d	window=%p\n", fullscreen, orient, window);
+	return -EINVAL; // NOT IMPLEMENTED
+}
+
+
+static void omx_vidisp_hide(struct vidisp_st *st)
+{
+	error("omx_vidisp_hide not implemented\n");
+}
```

Comment:
update-handler and hide-handler is optional, you can send in NULL instead of vidisp_register.